Libraries/Network/RCTHTTPRequestHandler.mm

```diff
@@ -60,10 +60,23 @@ - (NSURLSessionDataTask *)sendRequest:(NSURLRequest *)request
 {
   // Lazy setup
   if (!_session && [self isValid]) {
+    // You can override default NSURLSession instance property allowsCellularAccess:YES value by providing
+    // the following key to your RN project (either inside ios/Info.plist file or by editing it in Xcode):
+    // <key>ReactNetworkConfigChoice</key>    <string>OnlyUseWifi</string>
+    // This will set allowsCellularAccess to NO and force Wifi only for all network calls on iOS
+    // If you do not want to override default behavior, do nothing or set key with value "UseWifiAndCellular"
+    NSDictionary *infoDictionary = [[NSBundle mainBundle] infoDictionary];
+    NSString *customOverrideKey = [infoDictionary objectForKey:@"ReactNetworkConfigChoice"];
```

Comment:
Perhaps we should change this to RN_NETWORK_WIFI_ONLY: true / false

Reply:
I think that it should be clear upfront what the key does without opening it, since it’s a Boolean toggle. And using a string to compare doesn’t seem super helpful either.

Reply:
I don't think user defined tags can use the BOOL <true/> format(see [this S.O.](https://stackoverflow.com/questions/1832157/setting-a-boolean-property-in-info-plist-from-a-user-defined-setting) for more info). However, we could do something similar by making the string be YES or NO, but would still need to either check the string value or typecast it into a BOOL value, which I don't think would simplify things.

Reply:
I've updated the key to be ReactNetworkForceWifiOnly to try and match naming conventions and am now using a string value of YES. If you prefer to actually check a boolean value we would need to typecast this as Apple doesn't seem to allow user defined plist values to be booleans. 

I think it should be good to go as is and converting would add in an unnecessary step.
Regarding your offline question regarding isEqualToString, Apple has the following to say: "For values retrieved from external sources, such as an application’s information property list (Info.plist) or preferences, the use of isEqual: is preferred because it is safer; when the types are known, use isEqualToString: instead." [Source at the bottom](https://developer.apple.com/library/archive/documentation/Cocoa/Conceptual/CodingGuidelines/Articles/FrameworkImpl.html#//apple_ref/doc/uid/20001286-1007347)

Since we should know what the types are here sticking with isEqualToString seems to make the most sense to me. We could use isEqual instead of isEqualToString, but the additional comment of "By default isEqual: looks for pointer equality of each object’s address" makes me wonder if more is going on there than I understand and I would prefer feedback from someone who is more familiar with Objective C before changing that. It looks like we could coerce it into being [truthy](https://developer.apple.com/documentation/foundation/nsuserdefaults/1416388-boolforkey?language=objc), but I'd rather not be clever if possible.